src/helpers.cc

```diff
@@ -646,3 +646,24 @@ bool known_local_target(LiteClient& client, const Uptane::Target& t, std::vector
   }
   return known_target;
 }
+
+bool match_target_base(const Uptane::Target& t1, const Uptane::Target& t2) {
+  if (t1.type() != t2.type()) {
+    return false;
+  }
+
+  if (t1.length() != t2.length()) {
+    return false;
+  }
+
+  if (t1.filename() != t2.filename()) {
+    return false;
+  }
+
+  // match just OSTREE commit
+  if (t1.sha256Hash() != t2.sha256Hash()) {
```

Comment:
This is exactly how i'd always planned to implement this. I have wondered if making it generic like the upstream version does to account for sha512 might make sense. However, ostree can't do sha512, right? So I *think* the simplicity of this approach is better.

Reply:
yes, just KISS